### PR TITLE
feat: Expose a formatting method for routing header values

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: Google.Api.Gax.Grpc/ProtobufUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        value is 0 || value is 0L ||
+        // uint32, fixed32, uint64, fixed64
+        value is UnsignedInt32Zero || value is 0UL ||
+        // bytes
+        (value is ByteString bs && bs.IsEmpty) ||
+        // Any enum (all protobuf enums have an underlying type of Int32)
+        (value is Enum enumValue && Convert.ToInt32(enumValue) == 0);
+
+    /// <summary>
+    /// Formats a value in a way that is suitable for a header, URL path segment (after URL-encoding), or query parameter. The value
```
Comment (amanda-tarafa) [inline Google.Api.Gax.Grpc/ProtobufUtilities.cs:50]: can't comment here but, where it says `after URL-encoding` did you mean `before URL-encoding`?

These are not encoded, right?

Comment (amanda-tarafa) [reply Google.Api.Gax.Grpc/ProtobufUtilities.cs:50]: Ah, I think you meant it **will** be valid after URL-encoding.

File: Google.Api.Gax.Grpc/ProtobufUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    /// is effectively formatted the same way it is in the JSON representation. The return value of this
+    /// method is only guaranteed for single primitive values - not repeated fields, maps, or messages.
```
Comment (amanda-tarafa) [inline Google.Api.Gax.Grpc/ProtobufUtilities.cs:52]: Maybe if this is made public at some point, it might be worth it to fail for repeated, maps, etc.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: Google.Api.Gax.Grpc/ProtobufUtilities.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Gax.Grpc/ProtobufUtilities.cs:50]: Yes, that's right :)

File: Google.Api.Gax.Grpc/ProtobufUtilities.cs (hunk 2 above)
Comment (jskeet) [reply Google.Api.Gax.Grpc/ProtobufUtilities.cs:52]: Yup, potentially. I think we'd want to guarantee more stuff if we use it from anything that's expected to be used from hand-written code. I don't mind too much while it's just generated code that calls "things that call this".

**Pull request merged**